### PR TITLE
[FXML-4281] Remove flag for rounding mode of casting ops

### PR DESCRIPTION
**Pull request by cferry-AMD**
Per [upstream review](https://github.com/llvm/llvm-project/pull/87614), it appeared that the rounding mode does not affect casts but only function calls. Because we're only issuing casts in this pass, we can get rid of the `float-to-int-truncates` flag -- for casts, [this is always true](https://en.cppreference.com/w/cpp/numeric/fenv/FE_round).

### CODE REVIEW

**Review by TinaAMD — APPROVED** [review]
Nice that it was caught when upstreaming and thank you for following up!

**Review by mgehre-amd — APPROVED** [review]

File: mlir/include/mlir/Conversion/Passes.td [hunk 1]
```diff
@@ -139,24 +139,7 @@ def ArithToAMDGPUConversionPass : Pass<"convert-arith-to-amdgpu"> {
 
 def ConvertArithToEmitC : Pass<"convert-arith-to-emitc"> {
   let summary = "Convert Arith dialect to EmitC dialect";
-  let description = [{
-    This pass converts `arith` dialect operations to `emitc`.
```
Comment (mgehre-amd) [inline mlir/include/mlir/Conversion/Passes.td:143]: We should keep a short description

**Review by josel-amd — APPROVED** [review]

File: mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp [hunk 2]
```diff
@@ -317,14 +317,9 @@ class SelectOpConversion : public OpConversionPattern<arith::SelectOp> {
...
-private:
-  bool floatToIntTruncates;
-
 public:
-  FtoICastOpConversion(const TypeConverter &typeConverter, MLIRContext *context,
-                       bool optionFloatToIntTruncates)
-      : OpConversionPattern<CastOp>(typeConverter, context),
-        floatToIntTruncates(optionFloatToIntTruncates) {}
+  FtoICastOpConversion(const TypeConverter &typeConverter, MLIRContext *context)
+      : OpConversionPattern<CastOp>(typeConverter, context) {}
```
Comment (mgehre-amd) [inline mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp:322]:
```suggestion
  using OpConversionPattern::OpConversionPattern;
```
should work instead (maybe with template args).

**Review by mgehre-amd — APPROVED** [review]

File: mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp (hunk 2 above)
Comment (cferry-AMD) [reply mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp:322]: Sure! Thanks for picking this up :)

File: mlir/include/mlir/Conversion/Passes.td (hunk 1 above)
Comment (cferry-AMD) [reply mlir/include/mlir/Conversion/Passes.td:143]: I restored it but isn't it redundant with the summary? I don't know where this is actually used (doxygen maybe?).

Comment (mgehre-amd) [reply mlir/include/mlir/Conversion/Passes.td:143]: Not sure if there is a policy. I was under the impression that all passes have a description...?

Comment (cferry-AMD) [reply mlir/include/mlir/Conversion/Passes.td:143]: I honestly don't know either. That one currently doesn't have a description upstream.

Comment (TinaAMD) [reply mlir/include/mlir/Conversion/Passes.td:143]: I think this is a remainder of the PR I raised upstream to introduce ArithToEmitC, but then the other versioning was merged, so only we have the description.

Comment (mgehre-amd) [reply mlir/include/mlir/Conversion/Passes.td:143]: Sounds good to go with what upstream has

**Comment by cferry-AMD** [comment]
Reconciled this PR with what's been merged upstream.

**Pull request merged**